zerver/tests/test_service_bot_system.py

```diff
@@ -604,3 +606,27 @@ def test_no_trigger_on_group_direct_message_from_bot(
         recipients = [self.user_profile, self.bot_profile]
         self.send_group_direct_message(sender, recipients)
         self.assertFalse(mock_queue_event_on_commit.called)
+
+    @responses.activate
+    def test_flag_messages_outgoing_webhook_bot_has_processed(self) -> None:
+        """
+        Verifies that once an event has been processed by the outgoing webhook
+        bot's queue processor, the message is marked as processed (flagged with `read`).
+        """
+        self.bot_profile.bot_type = UserProfile.OUTGOING_WEBHOOK_BOT
+        self.bot_profile.save()
+        sender = self.user_profile
+        recipients = [self.user_profile, self.bot_profile, self.second_bot_profile]
+        responses.add(
+            responses.POST,
+            "https://bot.example.com/",
+            json="",
+        )
+        message_id = self.send_group_direct_message(
+            sender, recipients, content=f"@**{self.bot_profile.full_name}** foo"
+        )
+        # message = Message.objects.get(id=message_id, sender=sender)
+        bot_user_message = UserMessage.objects.get(
+            user_profile=self.bot_profile, message=message_id
+        )
+        self.assertIn("read", bot_user_message.flags_list())
```

Comment:
Ideally, we'd extend these tests to check that it wasn't marked as read until the outgoing webhook worker ran. Might be a nice follow-up if you can figure out an easy way to do that.